In [ ]:
import pandas as pd

df = pd.read_csv("C:\\Users\\tempu\\Downloads\\research\\labs\\gratton\\Arousal-Project\\splitting_sample\\iNET_sleep_scores_and_times.csv")
df.head()

,subject,session,run,task,file,path,FD,sleep.score,time
0,INET016,1,1,rest,sub-INET016_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,263,4.0,4.821667
1,INET016,1,2,rest,sub-INET016_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,259,4.0,4.748333
2,INET016,1,3,rest,sub-INET016_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,237,4.0,4.345000
3,INET016,1,4,rest,sub-INET016_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,257,4.0,4.711667
4,INET016,1,5,rest,sub-INET016_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,265,5.0,4.858333


In [10]:
df_eyes = pd.read_csv("C:\\Users\\tempu\\Downloads\\research\\labs\\gratton\\Arousal-Project\\splitting_sample\\iNETs-sleep-data.csv")
df_eyes["category"].value_counts()

category
eyes_closed    121
movement        79
Name: count, dtype: int64

There are only 121 eyes_closed runs. How many subjects are there?

In [18]:
df_eyes.head()

,file,subject,session,task,run,notes,sleep-score,FD-hold,category
0,sub-INET168_ses-4_task-rest_run-01_bold.nii.gz,INET168,4,rest,1,NaN,1,NaN,NaN
1,sub-INET168_ses-4_task-rest_run-02_bold.nii.gz,INET168,4,rest,2,NaN,1,NaN,NaN
2,sub-INET168_ses-4_task-rest_run-03_bold.nii.gz,INET168,4,rest,3,NaN,1,NaN,NaN
3,sub-INET168_ses-4_task-rest_run-04_bold.nii.gz,INET168,4,rest,4,NaN,2,NaN,NaN
4,sub-INET168_ses-4_task-rest_run-05_bold.nii.gz,INET168,4,rest,5,NaN,2,NaN,NaN


In [21]:
df_eyes_subj = df_eyes[df_eyes["category"] == "eyes_closed"].groupby("subject").agg("count").reset_index()
df_eyes_subj

,subject,file,session,task,run,notes,sleep-score,FD-hold,category
0,INET001,1,1,1,1,1,1,1,1
1,INET003,5,5,5,5,5,5,5,5
2,INET005,1,1,1,1,1,1,1,1
3,INET019,1,1,1,1,1,1,0,1
4,INET026,1,1,1,1,1,1,1,1
5,INET032,1,1,1,1,1,1,1,1
6,INET046,1,1,1,1,1,1,1,1
7,INET048,1,1,1,1,1,1,1,1
8,INET059,1,1,1,1,1,1,0,1
9,INET063,1,1,1,1,1,1,1,1


There are 39 subjects with at least 1 eyes_closed run. I just have to find out which of these eyes-closed subjects are precision.  

I need that one data frame with the inet times in it.

In [33]:
df_eyes = df_eyes.drop(columns = ["file", "task", "notes", "sleep-score"])
df_eyes.head(5)

,subject,session,run,FD-hold,category
0,INET168,4,1,NaN,NaN
1,INET168,4,2,NaN,NaN
2,INET168,4,3,NaN,NaN
3,INET168,4,4,NaN,NaN
4,INET168,4,5,NaN,NaN


In [34]:
df = df.merge(df_eyes, on=["subject", "session", "run"], how="left")
df.head()

,subject,session,run,task,file,path,FD,sleep.score,time,FD-hold,category
0,INET016,1,1,rest,sub-INET016_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,263,4.0,4.821667,NaN,NaN
1,INET016,1,2,rest,sub-INET016_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,259,4.0,4.748333,NaN,NaN
2,INET016,1,3,rest,sub-INET016_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,237,4.0,4.345000,NaN,NaN
3,INET016,1,4,rest,sub-INET016_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,257,4.0,4.711667,NaN,NaN
4,INET016,1,5,rest,sub-INET016_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,265,5.0,4.858333,NaN,NaN


I have made the master df with subject, session, run, task, file/path, DF, sleep.score, time, FD-hold, and category.  

I need to split the dfs by eyes_closed and eyes open (NaN), then group by subjects and aggregate with sum to figure out how much total time each subject has in each condition.

In [58]:
df_eo_sum = df[df["category"].isna()].groupby("subject").agg("sum").reset_index()
df_eo_sum.head()

,subject,session,run,task,file,path,FD,sleep.score,time,FD-hold,category
0,INET016,78,138,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET016_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,7584,126.0,139.040000,0,0
1,INET018,81,139,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET018_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,8179,97.0,149.948333,0,0
2,INET019,82,146,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET019_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,7808,47.0,143.146667,0,0
3,INET027,83,139,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET027_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,7996,80.0,146.593333,0,0
4,INET029,87,162,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET029_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,8422,0.0,154.403333,0,0


In [59]:
len(df_eo_sum)

90

90 subjects in high arousal (eo) group level analysis.

In [72]:
df_eo_prec = (df_eo_sum[df_eo_sum.time > 20]).reset_index()
df_eo_prec.head()

,index,subject,session,run,task,file,path,FD,sleep.score,time,FD-hold,category
0,0,INET016,78,138,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET016_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,7584,126.0,139.040000,0,0
1,1,INET018,81,139,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET018_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,8179,97.0,149.948333,0,0
2,2,INET019,82,146,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET019_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,7808,47.0,143.146667,0,0
3,3,INET027,83,139,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET027_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,7996,80.0,146.593333,0,0
4,4,INET029,87,162,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET029_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,8422,0.0,154.403333,0,0


In [78]:
len(df_eo_prec)

90

7 subjects in precision high arousal analysis (eo).

In [43]:
df_ec_sum = df[df["category"] == "eyes_closed"].groupby("subject").agg("sum").reset_index()
df_ec_sum.head()

,subject,session,run,task,file,path,FD,sleep.score,time,FD-hold,category
0,INET019,1,1,rest,sub-INET019_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,258,4.0,4.730000,0,eyes_closed
1,INET032,3,2,rest,sub-INET032_ses-3_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,247,3.0,4.528333,0.91111111111111098,eyes_closed
2,INET046,2,5,rest,sub-INET046_ses-2_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,257,6.0,4.711667,0.95185185185185195,eyes_closed
3,INET048,2,2,rest,sub-INET048_ses-2_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,264,4.0,4.840000,0.97777777777777797,eyes_closed
4,INET059,1,2,rest,sub-INET059_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,245,5.0,4.491667,0,eyes_closed


In [57]:
len(df_ec_sum)

36

For group analysis, I will have 36 subjects in low arousal (ec) condition.

In [75]:
df_ec_prec = (df_ec_sum[df_ec_sum.time > 20]).reset_index()
df_ec_prec.head()

,index,subject,session,run,task,file,path,FD,sleep.score,time,FD-hold,category
0,6,INET077,17,36,restrestrestrestrestrestrestrestrestrest,sub-INET077_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,2197,21.0,40.278333,0.977777777777777970.922222222222222050.933333...,eyes_closedeyes_closedeyes_closedeyes_closedey...
1,9,INET083,13,29,restrestrestrestrestrest,sub-INET083_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,1512,33.0,27.720000,0.855555555555555960.896296296296296040.974074...,eyes_closedeyes_closedeyes_closedeyes_closedey...
2,13,INET091,32,50,restrestrestrestrestrestrestrestrestrest,sub-INET091_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,2117,49.0,38.811667,0.677777777777778030.622222222222222010.707407...,eyes_closedeyes_closedeyes_closedeyes_closedey...
3,17,INET098,34,47,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET098_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,3216,31.0,58.960000,0,eyes_closedeyes_closedeyes_closedeyes_closedey...
4,21,INET107,16,15,restrestrestrestrest,sub-INET107_ses-2_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,1253,10.0,22.971667,0.903703703703704010.974074074074073980.874074...,eyes_closedeyes_closedeyes_closedeyes_closedey...


In [76]:
len(df_ec_prec)

7

I will have 7 precision subjects in low arousal precision analysis (ec). Now I need to find which ones have both high and kow arousal for precision amounts.

In [80]:
df_prec = df_ec_prec.merge(df_eo_prec, on=["subject"], how="inner", suffixes=("_ec", "_eo"))
df_prec.head()

,index_ec,subject,session_ec,run_ec,task_ec,file_ec,path_ec,FD_ec,sleep.score_ec,time_ec,...,session_eo,run_eo,task_eo,file_eo,path_eo,FD_eo,sleep.score_eo,time_eo,FD-hold_eo,category_eo
0,6,INET077,17,36,restrestrestrestrestrestrestrestrestrest,sub-INET077_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,2197,21.0,40.278333,...,50,73,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET077_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,4022,23.0,73.736667,0.9074074074074070.807407407407407020.72962962...,0
1,9,INET083,13,29,restrestrestrestrestrest,sub-INET083_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,1512,33.0,27.720000,...,61,91,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET083_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,5975,73.0,109.541667,0.974074074074073980.966666666666667010.974074...,0
2,13,INET091,32,50,restrestrestrestrestrestrestrestrestrest,sub-INET091_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,2117,49.0,38.811667,...,32,45,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET091_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,3407,48.0,62.461667,0.525925925925926020.825925925925925950.933333...,0
3,17,INET098,34,47,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET098_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,3216,31.0,58.960000,...,36,65,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET098_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,3220,28.0,59.033333,0,0
4,21,INET107,16,15,restrestrestrestrest,sub-INET107_ses-2_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,1253,10.0,22.971667,...,52,86,restrestrestrestrestrestrestrestrestrestrestre...,sub-INET107_ses-1_task-rest_desc-framenums_fFD...,/Volumes/illinois-las-psych-gratton/iNetworks/...,5116,46.0,93.793333,0.9370370370370370.977777777777777970.47407407...,0


# REPORT for 6/2

**Group Level Summary:**
- 90 subjects in high arousal
- 36 subjects in low arousal
- \>5 min of data for each subject

**Precision Sample Summary:**
- INET077, INET083, INET091, INET098, and INET107
- \>20 min of data for each subject

**Goals for 6/3**
1. Find which sessions/runs are usable in Poldrack's data and add to precision sample.
2. Match age and gender for INET subjects so that there are 36 subjects in each condition for group-level analyses. (Make sure no subject is repeated across conditions!)

### MyConnectome Data

The relevant variable is "morning:Sleepquality".

In [29]:
df_myc = pd.read_csv("C:\\Users\\tempu\\Downloads\\research\\labs\\gratton\\Arousal-Project\\splitting_sample\\trackingdata_poldrack.txt", sep="\t")
df_myc.head(5)

,subcode,date,afterscan:Anxietyduringscan,afterscan:diastolic,afterscan:pulse,afterscan:systolic,blood:ba,blood:eo,blood:hgb,blood:ly,...,scan:noisecancel,weather:precip,weather:temphi,weather:templo,weight,zeo:timeInDeep,zeo:timeInLight,zeo:timeInRem,zeo:totalZ,zeo:zq
0,sub001,2012-09-24,.,.,.,.,.,.,.,.,...,0,0.00,92,69,149.38,100,245,105,450,90
1,sub002,2012-09-25,.,.,.,.,.,.,.,.,...,0,0.00,92,71,149.03,76,287,101,464,88
2,sub003,2012-09-26,3,85,.,121,.,.,.,.,...,0,0.00,92,72,148.01,70,238,86,393,74
3,sub004,2012-09-27,3,69,.,114,.,.,.,.,...,0,0.00,90,69,148.01,.,.,.,.,.
4,sub005,2012-09-28,2,73,.,133,.,.,.,.,...,0,0.31,86,72,148.01,.,.,.,.,.


In [31]:
df_myc = df_myc[~df_myc["morning:Sleepquality"].isna()]
df_myc.head(5)

,subcode,date,afterscan:Anxietyduringscan,afterscan:diastolic,afterscan:pulse,afterscan:systolic,blood:ba,blood:eo,blood:hgb,blood:ly,...,scan:noisecancel,weather:precip,weather:temphi,weather:templo,weight,zeo:timeInDeep,zeo:timeInLight,zeo:timeInRem,zeo:totalZ,zeo:zq
0,sub001,2012-09-24,.,.,.,.,.,.,.,.,...,0,0.00,92,69,149.38,100,245,105,450,90
1,sub002,2012-09-25,.,.,.,.,.,.,.,.,...,0,0.00,92,71,149.03,76,287,101,464,88
2,sub003,2012-09-26,3,85,.,121,.,.,.,.,...,0,0.00,92,72,148.01,70,238,86,393,74
3,sub004,2012-09-27,3,69,.,114,.,.,.,.,...,0,0.00,90,69,148.01,.,.,.,.,.
4,sub005,2012-09-28,2,73,.,133,.,.,.,.,...,0,0.31,86,72,148.01,.,.,.,.,.
